automation/patch_igz/patch_remote.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/env python3
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import logging
+import os
+import re
+import subprocess
+import sys
+from contextlib import contextmanager
+from typing import List
+
+import click
+import coloredlogs
+import paramiko
+import yaml
+
+log_level = logging.INFO
+fmt = "%(asctime)s %(levelname)s %(message)s"
+logging.basicConfig(level=log_level)
+logger = logging.getLogger("mlrun-deploy")
+coloredlogs.install(level=log_level, logger=logger, fmt=fmt)
+
+
+class MLRunPatcher(object):
+    class Consts(object):
+        mandatory_fields = {"DATA_NODES", "SSH_USER", "SSH_PASSWORD", "DOCKER_REGISTRY"}
+
+    def __init__(self, conf_file):
+        self._config = yaml.safe_load(conf_file)
+        self._validate_config()
+
+    def patch_mlrun_api(self):
+        branch = re.sub(r"\\|@|/|:|#|~|\+", "_", self._find_current_git_branch())
+        vers = self._get_current_version()
+
+        nodes = self._config["DATA_NODES"]
+        if not isinstance(nodes, list):
+            nodes = [nodes]
+
+        image_tag = self._get_image_tag(branch, vers)
+        self._make_mlrun_api(image_tag)
+        built_image = "{}/mlrun-api:{}".format(
+            self._config["DOCKER_REGISTRY"], image_tag
+        )
```

Comment:
If the image is built in `_make_mlrun_api`, I'd expect it to return the new image tag instead of computing it after the fact. So like `built_image_tag =  self._make_mlrun_api(image_tag)`
Also if you want (not necessary), you can:
```
new_image_tag = f'{self._config["DOCKER_REGISTRY"]}/mlrun-api:{image_tag}'